### PR TITLE
Add Update query tests

### PR DESCRIPTION
**Pull request by Phillammon**
## What is the goal of this PR?

We added a new query type, so we needed to add tests to cover this new query type (or at least, the parts unique to it, as match, insert and delete have fairly good coverage of it).

## What are the changes implemented in this PR?

Added `update.feature`

### CODE REVIEW

File: graql/language/update.feature [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+#noinspection CucumberUndefinedStep
+Feature: Graql Delete Query
```
Comment (jmsfltchr) [inline graql/language/update.feature:19]: `Graql Update Query`

File: graql/language/update.feature [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    Given connection close all sessions
+    Given connection open data session for database: grakn
+    Given session opens transaction of type: write
+
+
+  ##########
+  # THINGS #
+  ##########
+
+  Scenario: simple update
```
Comment (jmsfltchr) [inline graql/language/update.feature:57]: Capitalise first word

File: graql/language/update.feature [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    Given transaction commits
+
+    Given connection close all sessions
+    Given connection open data session for database: grakn
+    Given session opens transaction of type: write
+
+
+  ##########
+  # THINGS #
+  ##########
```
Comment (jmsfltchr) [inline graql/language/update.feature:55]: If this comment for consistency with elsewhere? If not it's meaningless as we're always talking things.

File: graql/language/update.feature (hunk 2 above)
Comment (jmsfltchr) [reply graql/language/update.feature:57]: Also this scenario does more, it updates one instance without side effects on the other, so try to put that in the title.

File: graql/language/update.feature [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    When graql update; throws exception
+      """
+      match
+        $x isa person, has name "Alex";
+        $y isa person, has name "Bob";
+        $r (friend: $x) isa friendship;
+      delete
+        $r (friend: $x);
+      insert
+        $r (friend: $y);
```
Comment (jmsfltchr) [inline graql/language/update.feature:109]: Good test case, are there any other invalid states we could put ourselves into? We could take inspiration from our other existing tests.

File: graql/language/update.feature [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+      """
+      match
+        $p isa person, has name $n;
+        $nc isa nameclass, has name $n;
+      delete
+        $p has $n;
+      insert
+        (named: $p, name: $nc) isa naming;
+      """
+    Then transaction commits
```
Comment (jmsfltchr) [inline graql/language/update.feature:178]: Isn't there a value to doing the match and checking the answers? Otherwise how do we know this was successful?

**Review by jmsfltchr — CHANGES_REQUESTED** [review]

**Review by haikalpribadi — CHANGES_REQUESTED** [review]
I'm leaving the detailed review of the scenarios to @jmsfltchr, so I'm just commenting on the overall formatting: if there are multiple lines of `match/delete/insert` variables, put them on a new line but at the same indentation, not deeper. If not multiple lines keep them on the same line as `match/delete/insert` keyword.

**Review by jmsfltchr — APPROVED** [review]

**Review by haikalpribadi — APPROVED** [review]

**Pull request merged**